tools/boost-builder/Dockerfile

```diff
@@ -0,0 +1,9 @@
+# Does this match Travis's trusty?
+FROM ubuntu:trusty
```

Comment:
Changed tags from trusty to 14.04.5, which does match travis exactly. (Confirmed based on output at the beginning of the CI run for this PR.)